src/validators/schema-validator.ts

```diff
@@ -0,0 +1,139 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { OHRIFormSchema, ConfigObject } from '../api/types';
+
+export const handleFormValidation = async (
+  schema: OHRIFormSchema,
+  configObject: ConfigObject,
+): Promise<Array<Array<Record<string, any>>>> => {
+  const errors = [];
+  const warnings = [];
+
+  if (schema) {
+    const asyncTasks = [];
+
+    schema.pages?.forEach(page =>
+      page.sections?.forEach(section =>
+        section.questions?.forEach(question => {
+          asyncTasks.push(
+            handleQuestionValidation(question, errors, warnings, configObject),
+            handleAnswerValidation(question, errors),
+          );
+          question.type === 'obsGroup' &&
+            question.questions?.forEach(obsGrpQuestion =>
+              asyncTasks.push(
+                handleQuestionValidation(obsGrpQuestion, errors, configObject, warnings),
+                handleAnswerValidation(question, errors),
+              ),
+            );
+        }),
+      ),
+    );
+    await Promise.all(asyncTasks);
+
+    return [errors, warnings];
+  }
+};
+
+const handleQuestionValidation = async (conceptObject, errorsArray, warningsArray, configObject) => {
+  const conceptRepresentation =
+    'custom:(uuid,display,datatype,answers,conceptMappings:(conceptReferenceTerm:(conceptSource:(name),code)))';
+
+  const searchRef = conceptObject.questionOptions.concept
+    ? conceptObject.questionOptions.concept
+    : conceptObject.questionOptions.conceptMappings?.length
+    ? conceptObject.questionOptions.conceptMappings
+        ?.map(mapping => {
+          return `${mapping.type}:${mapping.value}`;
+        })
+        .join(',')
+    : '';
+
+  if (searchRef) {
+    try {
+      const { data } = await openmrsFetch(`/ws/rest/v1/concept?references=${searchRef}&v=${conceptRepresentation}`);
+      if (data.results.length) {
+        const [resObject] = data.results;
+
+        resObject.datatype.name === 'Boolean' &&
+          conceptObject.questionOptions.answers.forEach(answer => {
+            if (
+              answer.concept !== 'cf82933b-3f3f-45e7-a5ab-5d31aaee3da3' &&
+              answer.concept !== '488b58ff-64f5-4f8a-8979-fa79940b1594'
```

Comment:
I would do something like:

```js

if ( ![ConceptTrue, ConceptFalse].includes(answer.concept) ) {
   // handle error
}

```

The ConceptTrue and ConceptFalse are driven by a System configuration: `concept.true` and `concept.false`. We need to be picking these values from the backend. On the contrary, these are currently [defined as constants](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/constants.ts#L1-L2) within the library which isn't the conventional practice. I've always wanted to address this technical debt but never got the latitude. 

Ideally, we should introduce a new hook that will read this system setting. It would be great if you offered to address this otherwise you can use the already defined constants and we address this later on.

Reply:
This, is well noted. I would also like to take on the issue of the hook reading the values from the backend if possible.